### DISCUSSION TITLE
Improve search performance

### DISCUSSION THREAD

**Discussion by Makulak** [directus/directus#16768] [Archive]
Hi,

we have quite big table, with around 100 columns, and we have a problem with searching items - it takes a really long time to filter records using just string input, as it searches through all columns.

I know, that there's an option to specify columns and conditions to search, but for end-user is not as convenient as just typing text into field.

So - is there any way to select columns, that are used in text search?
Ideal solution would be to have list of 'searchable' columns to set somewhere in settings. 
Maybe You have other solutions and ideas how to improve search performance?

**Comment by paescuj** [comment]
Might be relevant: https://github.com/directus/directus/discussions/12264
